### PR TITLE
add optional argument to stop maintaining team members, refs #65

### PR DESCRIPTION
**Pull request by arnisoph**

### CODE REVIEW

**Comment by arnisoph** [comment]
Okay, apparently this already works now for me in production. Ready for review & feedback. :)

File: opsgenie/resource_opsgenie_team.go [hunk 1]
```diff
@@ -63,10 +69,15 @@ func resourceOpsGenieTeamCreate(d *schema.ResourceData, meta interface{}) error
...
 		Name:        name,
 		Description: description,
-		Members:     expandOpsGenieTeamMembers(d),
 	}
 
-	log.Printf("[INFO] Creating OpsGenie team '%s'", name)
+	if d.Get("manage_members").(bool) {
+		createRequest.Members = expandOpsGenieTeamMembers(d)
+	} else if len(d.Get("member").([]interface{})) > 0 {
+		return fmt.Errorf("you've set manage_members=false but added member resources blocks to team %q: %q", name, d.Get("member"))
```
Comment (arnisoph) [inline opsgenie/resource_opsgenie_team.go]: this validation doesn't  work yet

**Comment by arnisoph** [comment]
@ffahri et al. feedback?

**Comment by ffahri** [comment]
@arnisoph thank you for this contribution let me try on both ci and locally lgtm

**Comment by ffahri** [comment]
Its good but i think we dont set this to default value. Because if some users have a lot of teams in their tf configurations they need to update this after we release like this. How about we use something like this instead
```
{
 ignore_members=true
}
```
If we dont set this to a default value its false by default (as bool). And if we expect user to enter true in the file we can opt out team memberships like this
```
if len(d.Get("member").([]interface{})) > 0 && !d.Get("ignore_membership").(bool) {

```

This way people wont do anything when they upgraded their plugin version.

What you think about this @arnisoph

**Comment by ffahri** [comment]
I tested this on local and its g2g but like i said if we introduce the default value people need to reapply their configuration file (and they really don't need to because they didn't change anything in their configuration files)

**Comment by arnisoph** [comment]
The behaviour you described was already implemented. :D I just renamed the attr and inverted bool-related logic.

Hope it's fine now. :)

**Comment by ffahri** [comment]
@arnisoph you right but we forgot this case :D we should not troll anyone with new release.

travis shows red could you please look at this

**Comment by ffahri** [comment]
we will send this release after this ci flow and we can expect other fields with v0.3 version

**Comment by arnisoph** [comment]
already done.  :)

File: opsgenie/resource_opsgenie_team_test.go [hunk 2]
```diff
@@ -156,6 +258,22 @@ resource "opsgenie_team" "test" {
 `, rString)
 }
 
+func testAccOpsGenieTeam_basicNoMember(randomUser, rString string) string {
+	return fmt.Sprintf(`
+resource "opsgenie_team" "test" {
+  name           = "genieteam-%s"
+  description    = "This team deals with all the things"
+  ignore_members = false
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_team_test.go]: can. you update this field as well

Comment (ffahri) [reply opsgenie/resource_opsgenie_team_test.go]: otherwise our e2e test will not pass :)

**Review by ffahri — CHANGES_REQUESTED** [review]

File: opsgenie/resource_opsgenie_team_test.go (hunk 2 above)
Comment (ffahri) [reply opsgenie/resource_opsgenie_team_test.go]:
```
TestAccOpsGenieTeam_basicNoMember
  === RUN   TestAccOpsGenieTeam_basicNoMember
--- FAIL: TestAccOpsGenieTeam_basicNoMember (2.52s)
    testing.go:569: Step 0 error: After applying this step and refreshing, the plan was not empty:
        
        DIFF:
        
        UPDATE: opsgenie_team.test
          description:    "This team deals with all the things" => "This team deals with all the things"
          id:             "1616fef5-4fb2-43b3-baf0-deb36a7fc9c9" => "1616fef5-4fb2-43b3-baf0-deb36a7fc9c9"
          ignore_members: "false" => "false"
          member.#:       "1" => "0"
          member.0.id:    "18ab82fa-1461-4e19-80f3-ed5caf85cd6c" => ""
          member.0.role:  "user" => ""
          name:           "genieteam-mn6lfl" => "genieteam-mn6lfl"
        
        
        
        STATE:
        
        opsgenie_team.test:
          ID = 1616fef5-4fb2-43b3-baf0-deb36a7fc9c9
          provider = provider.opsgenie
          description = This team deals with all the things
          ignore_members = false
          member.# = 1
          member.0.id = 18ab82fa-1461-4e19-80f3-ed5caf85cd6c
          member.0.role = user
          name = genieteam-mn6lfl
        
          Dependencies:
            opsgenie_user.test
        opsgenie_user.test:
          ID = 18ab82fa-1461-4e19-80f3-ed5caf85cd6c
          provider = provider.opsgenie
          full_name = Acceptance Test User
          locale = en_US
          role = User
          timezone = America/New_York
          username = genietest-z7m04a@opsgenie.com
FAIL
  INFO[2020-03-19T10:13:28.562510501Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:28.569737798Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:28.570630182Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:29.660732008Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:29.71641125Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.05982477Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.126664758Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.196179068Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.336264972Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.346678315Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.347505211Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.415075811Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.485343337Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.500183214Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.501104884Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.553893579Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.621665678Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.622671852Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.843289972Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
INFO[2020-03-19T10:13:30.968690084Z] Client is configured with ApiUrl: api.opsgenie.com, LogLevel: info, RetryMaxCount: 10 
```

Comment (arnisoph) [reply opsgenie/resource_opsgenie_team_test.go]: fixed. my n-th opsgenie test account has expired so I did not re-restarted e2e tests.

**Review by ffahri — APPROVED** [review]

**Pull request merged**